### DISCUSSION TITLE
Security in Prisma

### DISCUSSION THREAD

**Discussion by MohamadrezaKazemian** [prisma/prisma#17010] [Q&A] [answered]
recently I have been questioning myself a lot about security in Prisma.
since it's very easy to learn and use Prisma I feel like I'm missing something like security in my database and my APIs! 
is there any way or anything to check the security of the Prisma project? what do you suggest to do or refuse to do?
thanks for your help!

**Comment by pantharshit00** [comment, marked as answer]
Hello,

I am not particularly sure what you want to ask here. Prisma is just a ORM which lets you access the database. Our source code is all open source and available on GitHub so anyone can audit it if they want to assess any security related issues. 

If you are asking for general advice for how you can build secure applications with Prisma, I will say you will need to follow general security recommendation like never trusting user input, validating everything and having frequent audit. Let me know if this answers your question.

**Reply by Showdown76py** [reply]
why am I receiveing this?

**Reply by pantharshit00** [reply]
I am not sure. Are you watching discussions of this repository in your notification settings?

**Reply by AdrianPaulCarrieres** [reply]
It's because we got mentioned (in this discussion) by a spam bot or something earlier today, GitHub made us automatically watch this discussion because of it

**Comment by MohamadrezaKazemian** [comment]
> Hello,
> 
> I am not particularly sure what you want to ask here. Prisma is just a ORM which lets you access the database. Our source code is all open source and available on GitHub so anyone can audit it if they want to access any security related issues. 
> 
> If you are asking for general advice for how you can build secure applications with Prisma, I will say you will need to follow general security recommendation like never trusting user input, validating everything and having frequent audit. Let me know if this answers your question. 

Well. One of the main problems i have is the authentication with Prisma. I wanna use Jwt and using has methods to save passwords.

**Reply by pantharshit00** [reply]
You can checkout this example: https://github.com/prisma/prisma-examples/tree/latest/typescript/graphql-auth

It uses GraphQL and JWTs. 

That being said you can following any Node.js tutorial on JWT and just replaces database calls in that with Prisma.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
